internal/pkg/warewulfd/api/api.go

```diff
@@ -73,6 +73,7 @@ func Handler(auth *config.Authentication, allowedNets []net.IPNet) *web.Service
 			r.Method(http.MethodPut, "/{name}", nethttp.NewHandler(createOverlay()))
 			r.Method(http.MethodDelete, "/{name}", nethttp.NewHandler(deleteOverlay()))
 			r.Method(http.MethodDelete, "/{name}/file", nethttp.NewHandler(deleteOverlayFile()))
+			r.Method(http.MethodPatch, "/{name}/file", nethttp.NewHandler(updateOverlay()))
```

Comment:
Since this always replaces the whole file, I _think_ this should be a PUT, not a PATCH.

Reply:
thanks! have updated.

Reply:
hmm, I am seeing other similar apis use `PATCH` method, not sure if we need to follow the convention.

Reply:
also I am creating another rest api using PUT method 
https://github.com/warewulf/warewulf/pull/1952

Reply:
> > I am seeing other similar apis use PATCH method, not sure if we need to follow the convention.
> 
> Can you give a specific example?

`update*` methods in `api.go`, for example, https://github.com/JasonYangShadow/warewulf/blob/create_overlay/internal/pkg/warewulfd/api/api.go#L60

Reply:
`updateNode`, `updateProfile`, and `updateImage` all allow the specification of partial data that is applied / merged into the targeted node, profile, or image. `updateOverlayFile`, on the other hand, by its nature replaces the entire contents of the file with what is presented. This is more indicative of a PUT than a PATCH.